client/js/Core/main.js

```diff
@@ -1,70 +1,75 @@
-var scHelper = null;
-var scKeynodes = null;
+const scHelper = null;
+const scKeynodes = null;
 const currentYear = new Date().getFullYear();
 
-SCWeb.core.Main = {
+function ScClientCreate() {
+    let res, rej;
+    let serverHost = "ws://localhost:8090/ws_json";
```

Comment:
We cannot afford to have it hardcoded, it would break projects in production if `sc-machine` instance is hosted elsewhere. Heck, it would even break `docker compose` usecases. We aren't running this server on `node`, so we cannot use `dotenv`. We also do not use Webpack (though at some point we really **have** to migrate from Grunt, but it would require some effort from the frontend team, and the priorities are different, bla-bla-bla). Considering all of the above, the best solution I can suggest right now is to configure it using Grunt. Say, you could grab the environment variable such as `SC_JSON_URL` by doing
```js
const env_vars = process.env;
const SC_JSON_URL = env_vars.SC_JSON_URL;
```

Then you could use something like [this Grunt plugin](https://www.npmjs.com/package/grunt-replace) ([this](https://github.com/ejfrancis/grunt-envpreprocess) could also work, but it only works with files, which is less than ideal) to replace the text in a source file at build time, something like:
```js
{
  options: {
    patterns: [
      {
        match: '@@sc_json_url',
        replacement: '<%= SC_JSON_URL %>'
      }
    ]
  },
  files: [
    {
      expand: true, flatten: true, src: ['client/js/Core/main.js'], dest: 'build/'
    }
  ]
}
```
Of course it'd be a bit more complicated than that (at least we have to configure the default value), but that's one of the possible ways to implement it without duct taping or rewriting the whole build system.

Reply:
@FallenChromium As I see, ``serverHost`` is just config of sc-web. And after applying your proposition, we will have two config for sc-web: in ``sc-web.ini`` and  in Grunt.

I think we need to manage this config in sc-web.ini. We can do it in another PR because this hard code was before this PR

Reply:
You're right! Sounds like a better idea